### ISSUE TITLE
Make global parameter settings (params.py) more user-configurable

### ISSUE THREAD

**Issue by etal** [etal/cnvkit#74]
`params.py` contains some hard-coded filter settings that generally should not be touched, but on other systems there might be some interest in changing these parameters. Currently this can only be done by editing that file in the installed code (or the source code, then reinstalling).

A standard config file makes sense, but distributing it could be painful. So, I'm open to suggestions.

**Comment by etal** [comment]
`READ_LEN` is just a scaling parameter used in the coverage calculation (since the output of `samtools bedcov` is the number of aligned bases in a range, not a read count). It's also used in `export theta`, again as an arbitrary scaling parameter, to estimate the total number of reads in a segment. In both cases only relative values are used in the downstream calculations, so getting this value right is just for fun. Future work might use read lengths more critically, though.

`INSERT_SIZE` is used in `antitarget` to leave a suitable margin around each target to avoid picking up sequence fragments that were captured. The margin is twice the insert size, since insert sizes are variable, and the default value is a generous estimate of what we see in FFPE samples. (Twice that is what we see in fresh-frozen samples.) If the actual insert size is much larger, then in reference-free calling the log2 values of antitargets adjacent to baits could be slightly inflated. If a pooled or paired reference is used, the effect would be normalized away.

In `fix`, the parameter is used to calculate the "edge" biases, i.e. roll-off in coverage near the edge of baited regions and extra reads captured in the regions immediately flanking the baits, which can overlap adjacent baits. If the parameter is wrong here then the effectiveness of the "edge" bias correction could be reduced (but not eliminated or reversed).

TL;DR: Pretty robust. The parameters are not critical and most users shouldn't worry about them too much. Future work could also consider sniffing these values from the input BAM file(s) and storing them somehow, maybe generating this config file automatically.

**Comment by kyleabeauchamp** [comment]
Cool.  Thanks again for the detailed description.

**Comment by JspSrs** [comment]
Hi Eric,

From a more distant point of view, I have noticed that some control on ‘internal processes’ for end-users helps in adopting packages and to truly incorporate into their analysis. At the same time, mostly the default values are used. A config-file does also help in version control at that institute or if the same package is used in easily switching between two analysis types (as long as local-config-files are allowed by the software). The latter would the help in circumventing distributing issues.

Extrapolating our experience, one does not want technicians to change software inside the actual package, or as minimal as possible. Changing config-files is however quite doable. However, I am in a mainly diagnostic orientated lab. Perhaps in pure research this is different, but also there not everybody is ad ease with software, except via a GUI.

Regards, Jasper
